java/server/src/com/appdynamics/wpt/SATaskerClient.java

```diff
@@ -0,0 +1,139 @@
+  private final String taskerBaseUrl;
+  private final Gson gson;
+
+  public SATaskerClient(String taskerBaseUrl, Logger logger) {
+    this.logger = logger;
+    this.taskerBaseUrl = taskerBaseUrl;
+    this.browserReadyResource = new HttpGet(taskerBaseUrl + "/v1/browser/ready");
+    this.webdriverActionResource = new HttpPost(taskerBaseUrl + "/v1/webdriver/action");
+    this.client = new HttpClientFactory().getHttpClient();
+    this.gson = new Gson();
+  }
+
+  public boolean isBrowserReady() throws IOException, HttpRequestException {
+    logger.info("Attempting GET /v1/browser/ready on " + taskerBaseUrl);
+    HttpResponse response;
+
+    try {
+      response = client.execute(browserReadyResource);
+    } catch (Exception e) {
+      logger.severe("Failed to connect to the synthetic tasker at: " + taskerBaseUrl);
+      throw new HttpRequestException("Error connecting to synthetic tasker", e);
+    }
+
+    int statusCode = response.getStatusLine().getStatusCode();
+    if (statusCode >= 400) {
+      logger.severe("Failed to retrieve browser readiness from the tasker. Status code: "
+                    + statusCode);
+      throw new HttpRequestException(statusCode);
+    }
+
+    String responseBody = CharStreams.toString(
+      new InputStreamReader(response.getEntity().getContent())
+    );
+
+    JsonObject browserReadiness = new JsonParser().parse(responseBody).getAsJsonObject();
+
+    if (browserReadiness != null) {
+      logger.info("  -> Response: " + responseBody);
+      if (browserReadiness.has("ready")) {
+        return browserReadiness.getAsJsonPrimitive("ready").getAsBoolean();
+      }
+    }
+
+    return false;
+  }
+
+  public void submitWebDriverAction(Command command) throws HttpRequestException {
+    logger.info("Sending next webdriver action: " + command.getName());
+    HttpResponse response;
+
+    SAWebDriverAction action = new SAWebDriverAction()
+      .withTimestampMs(System.currentTimeMillis())
+      .withCommand(command.getName());
+
+    String actionJson = gson.toJson(action);
+
+    webdriverActionResource.setHeader("Content-Type", "application/json");
+    webdriverActionResource.setEntity(new StringEntity(actionJson, ContentType.APPLICATION_JSON));
+
+    try {
+      response = client.execute(webdriverActionResource);
+    } catch (Exception e) {
+      logger.severe("Failed to connect to the synthetic tasker at: " + taskerBaseUrl);
+      throw new HttpRequestException("Error connecting to synthetic tasker", e);
+    }
+
+    int statusCode = response.getStatusLine().getStatusCode();
+    if (statusCode >= 400) {
+      logger.severe("Failed to retrieve browser readiness from the tasker. Status code: "
+                    + statusCode);
+      throw new HttpRequestException(statusCode);
+    }
+
+    logger.info("WebDriver action sent successfully. Status code: " + statusCode);
+  }
+
+  // TODO: Should try and re-use the one in synthetic-tasker-api after adding it as a dependency.
```

Comment:
This should probably be a ticket and removed from the code, otherwise it'll stay there forever